### PR TITLE
chore: improve logging around lifecycle scripts

### PR DESCRIPTION
**Pull request by johnstcn**
Closes https://github.com/coder/envbuilder/issues/418

- Adds improved logging around lifecycle scripts and whether we're building from a devcontainer.json or a Dockerfile
- Improves integration tests around lifecycle script execution

### CODE REVIEW

File: envbuilder.go [hunk 1]
```diff
@@ -169,6 +169,7 @@ func run(ctx context.Context, opts options.Options, execArgs *execArgsInfo) erro
 		RemoteEnv:    make(map[string]string),
 	}
 	if fileExists(opts.Filesystem, workingDir.Image()) {
+		opts.Logger(log.LevelInfo, "Found magic image file at %s", workingDir.Image())
```
Comment (mafredri) [inline envbuilder.go:172]: I'd have gone with debug, but I think your choice of info is better. This'll be helpful in understanding what type of image a user is running.

File: envbuilder.go [hunk 2]
```diff
@@ -334,6 +337,7 @@ func run(ctx context.Context, opts options.Options, execArgs *execArgsInfo) erro
 		} else {
 			// If a Dockerfile was specified, we use that.
 			dockerfilePath := filepath.Join(buildTimeWorkspaceFolder, opts.DockerfilePath)
+			opts.Logger(log.LevelInfo, "Using Dockerfile: %s", opts.DockerfilePath)
```
Comment (mafredri) [inline envbuilder.go:340]: How about something like "Building in Dockerfile-only mode"?

File: envbuilder.go [hunk 3]
```diff
@@ -1406,6 +1410,7 @@ func execOneLifecycleScript(
 	userInfo userInfo,
 ) error {
 	if s.IsEmpty() {
+		logf(log.LevelInfo, "=== no %s specified", scriptName)
```
Comment (mafredri) [inline envbuilder.go:1413]:
```suggestion
		logf(log.LevelInfo, "=== No %s specified", scriptName)
```

**Review by mafredri — APPROVED** [review]
V nice!

File: envbuilder.go (hunk 1 above)
Comment (johnstcn) [reply envbuilder.go:172]: Yeah it's way too easy to miss this otherwise!

File: envbuilder.go (hunk 2 above)
Comment (johnstcn) [reply envbuilder.go:340]: I'll update the corresponding line for "Devcontainer mode" also 👍

**Pull request merged**